appservice/src/intTest/groovy/nu/yona/server/OverwriteUserTest.groovy

```diff
@@ -1,220 +1,184 @@
+	{
 		given:
-		def userAddResponse = appService.addUser(userCreationJSON, "Password");
-		def overwriteRequestResponse = appService.requestOverwriteUser(userCreationMobileNumber)
-		def userURL = appService.stripQueryString(userAddResponse.responseData._links.self.href);
-		def confirmationCode = overwriteRequestResponse.responseData.confirmationCode;
+			def richardAndBob = addRichardAndBobAsBuddies()
+			def richard = richardAndBob.richard
+			def bob = richardAndBob.bob
+			def confirmationCode = appService.requestOverwriteUser(richard.mobileNumber)?.responseData?.confirmationCode
+			def richardChanged = appService.addUser(appService.&assertUserOverwriteResponseDetails, "${richard.password}Changed", "${richard.firstName}Changed",
+				"${richard.lastName}Changed", "${richard.nickname}Changed", richard.mobileNumber, ["Nokia"], ["news"],
+				["overwriteUserConfirmationCode": confirmationCode])
+			def buddy = appService.getBuddies(bob)[0]
 
 		when:
-		def response1TimeWrong = appService.addUser(userCreationJSON, "New password", ["overwriteUserConfirmationCode": "${confirmationCode}1"])
-		appService.addUser(userCreationJSON, "New password", ["overwriteUserConfirmationCode": "${confirmationCode}2"])
-		appService.addUser(userCreationJSON, "New password", ["overwriteUserConfirmationCode": "${confirmationCode}3"])
-		appService.addUser(userCreationJSON, "New password", ["overwriteUserConfirmationCode": "${confirmationCode}4"])
-		def response5TimesWrong = appService.addUser(userCreationJSON, "New password", ["overwriteUserConfirmationCode": "${confirmationCode}5"])
-		def response6TimesWrong = appService.addUser(userCreationJSON, "New password", ["overwriteUserConfirmationCode": "${confirmationCode}6"])
-		def response7thTimeRight = appService.addUser(userCreationJSON, "New password", ["overwriteUserConfirmationCode": "${confirmationCode}"])
+			def response = appService.removeBuddy(bob, buddy, "Good bye friend")
 
 		then:
-		confirmationCode != null
-		userAddResponse.status == 201
-		userAddResponse.responseData.mobileNumberConfirmed == false
-		response1TimeWrong.status == 400
-		response1TimeWrong.responseData.code == "error.mobile.number.confirmation.code.mismatch"
-		response5TimesWrong.status == 400
-		response5TimesWrong.responseData.code == "error.mobile.number.confirmation.code.mismatch"
-		response6TimesWrong.status == 400
-		response6TimesWrong.responseData.code == "error.too.many.wrong.attempts"
-		response7thTimeRight.status == 400
-		response7thTimeRight.responseData.code == "error.too.many.wrong.attempts"
+			response.status == 200
+			appService.getBuddies(bob).size() == 0
 
 		cleanup:
-		if (userURL) {
-			appService.deleteUser(userURL, "Password")
-		}
+			appService.deleteUser(richard)
+			appService.deleteUser(bob)
 	}
 
-	def 'Delete users'(){
+	def 'Hacking attempt: Brute force overwrite mobile number confirmation'()
+	{
+		given:
+			def userCreationMobileNumber = "+${timestamp}99"
+			def userCreationJSON = """{
```

Comment:
I'm not sure what you mean with this. This JSON fragment is only used here. Do you propose to move it out of this method?